32.md

```diff
@@ -0,0 +1,166 @@
+- `confidence` may have a value of 0 to 1. This indicates the certainty which the author has about their rating.
+- `context` may be a comma-separated list of urls (including NIP-21 urls) indicating other context that should be considered when interpreting labels.
+
+Content
+-------
+
+Labels should be short, meaningful strings. Longer discussions, such as for a review, or an
+explanation of why something was labeled the way it was, should go in the event's `content` field.
+
+Example events
+--------------
+
+A single event can apply multiple labels to multiple targets to support mass-tagging. Multiple
+namespaces may be used at the same time.
+
+```json
+{
+  "kind": 1985,
+  "tags": [
+    ["e", <id>, <relay_url>],
+    ["p", <id>, <relay_url>],
+    ["t", "chickens"],
+    ["L", "#t"]
+    ["L", "ugc"]
+    ["L", "com.example.labels"]
+    ["l", "chickens", "#t"],
+    ["l", "user generated content", "ugc"],
+    ["l", "permaculture", "com.example.labels"],
+    ["l", "permies", "com.example.labels"],
+    ["l", "farming", "com.example.labels"],
+  ],
+  "content": "",
+  ...
+}
+```
+
+A suggestion that multiple pubkeys be associated with the `permies` topic.
+
+```json
+{
+  "kind": 1985,
+  "tags": [
+    ["L", "#t"],
+    ["l", "permies", "#t"],
+    ["p", <pubkey1>, <relay_url>],
+    ["p", <pubkey2>, <relay_url>]
+  ],
+  "content": "",
+  ...
+}
+```
+
+A review of a relay, as relates to certain topics, including additional dimensions. The author
+is indicating here that `relay_url` is related to the bitcoin topic, but they're not very sure
+that's the case.
+
+```json
+{
+  "kind": 1985,
+  "tags": [
+    ["L", "#t"],
+    ["l", "bitcoin", "#t", "{\"quality\": 0.7, \"confidence\": 0.2}"],
+    ["r", <relay_url>]
+  ],
+  "content": "I think this relay is mostly just bitcoiners.",
+  ...
+}
+```
+
+A plain review of a relay.
+
+```json
+{
+  "kind": 1985,
+  "tags": [
+    ["L", "social.coracle.ontology"],
+    ["l", "review", "social.coracle.ontology", "{\"quality\": 0.1}"],
+    ["r", <relay_url>]
+  ],
+  "content": "This relay is full of mean people.",
+  ...
+}
+```
+
+A more abstract use case: defining an edge in a graph structure, in this case identifying
+a lightning channel that is open between two pubkeys. This just demonstrates the flexibility
+this spec provides for overlaying structured metadata on top of nostr.
+
+```json
+{
+  "kind": 1985,
+  "tags": [
+    ["L", "my-lightning-nomenclature"],
+    ["l", "channel", "my-lightning-nomenclature"],
+    ["p", <pubkey1>, <relay_url>],
+    ["p", <pubkey2>, <relay_url>]
+  ],
+  "content": "<channel_id>",
+  ...
+}
+```
+
+Publishers can self-label by adding `l` tags to their own non-1985 events.
+
+```json
+{
+  "kind": 1,
+  "tags": [
+    ["L", "social.nos.ontology"],
+    ["l", "IL-frd", "social.nos.ontology"]
+  ],
+  "content": "Send me 100 sats and I'll send you 200 back",
+  ...
+}
+```
```

Comment:
```suggestion
```

For simplicity, we should have a single way of doing things.

Reply:
I see your point - semantically these are two ways of arriving at the same state. However I think being able to label something in the original event is desirable, and for some use cases it's necessary.

The immediate need in my mind is using labels for content warnings. If I'm publishing a note with a content warning on it I don't want that label stored separately from the event. It creates lots of failure modes like: 
- some relays could store only the note and not the 1985 label event. Either accidentally or maliciously.
- clients that support content warnings would need to wait until all relays have confirmed that they don't have any associated 1985 events before displaying any note - creating a huge overhead. 

Content warnings are one case but there are others where having the label and event in the same place is desirable. For instance I could use labels to publish notes under different "streams": one for my notes on gardening, one for my notes about video games, etc. Being able to fetch the notes for a stream in a single REQ is much preferable to needing to fetch all the 1985 events for a stream and then individually request all the events in the stream.

Reply:
There is a NIP for content warnings already. Is it part of this NIP here that we will deprecate that one?

For "streams" shouldn't you be using `t` tags?

Reply:
This NIP informally intrudes on NIP 32 and 52's territory. The distinction is that 32 and 52 are designed for creating a very simple flag for clients/relays to manually moderate or hide stuff. This NIP *could* be used for that as well, but is intended to support more robust decentralized moderation — where users choose what vocabularies they care about. NIP 32 and 52 should still be used to flag stuff for centralized-style moderation or hiding in user feeds.

That's not to say I don't think 32 and 52 are bad and should be deprecated, but I don't want to force client authors to change how they do things if it's working for thm.

Reply:
I haven't seen the other NIP for content warnings. I'll check it out.

Reply:
NIP-36 doesn't have a space for a machine-readable category, which is important for how we want to implement them. So I'm keen to keep the self-tagging in this NIP. 

(Since NIP-36 is already being used by amethyst we'll probably use the `content-warning` tag in addition to `l` and `L` tags when publishing a note with a content warning for compatibility.)

Reply:
What about editing NIP-36 to add the machine-readable category on an optional tag item and link it to this NIP? I think that would be the best, unless I'm missing something.

Reply:
> What about editing NIP-36 to add the machine-readable category on an optional tag item and link it to this NIP? I think that would be the best, unless I'm missing something.

NIPs 36 & 56 don’t use single letter tags, hence they are searchable. Ergo neither really is capable of doing the job it needs to do.

Reply:
I think I get what @fiatjaf means, I've added a note to NIP 36 and 56.